src/field/scrcmd_mystery_gift.c

```diff
@@ -0,0 +1,684 @@
+#include "global.h"
+#include "bag.h"
+#include "mystery_gift.h"
+#include "scrcmd.h"
+#include "math_util.h"
+#include "save_vars_flags.h"
+#include "script_pokemon_util.h"
+#include "ribbon.h"
+#include "pokedex.h"
+#include "sys_vars.h"
+#include "fashion_case.h"
+#include "unk_020290B4.h"
+#include "pokewalker.h"
+#include "trainer_memo.h"
+#include "constants/map_sections.h"
+#include "msgdata/msg.naix"
+#include "msgdata/msg/msg_0209.h"
+
+struct GetMysteryGiftGmmState {
+    FieldSystem* fieldSys;
+    MessageFormat* msgFormat;
+    MysteryGiftData* mgData;
+};
+
+struct ScriptMysteryGiftFuncs {
+    BOOL (*check)(FieldSystem*, MysteryGiftData*);
+    void (*give)(FieldSystem*, MysteryGiftData*);
+    void (*messageSuccess)(struct GetMysteryGiftGmmState*, u16*, u16*);
+    void (*messageFailure)(struct GetMysteryGiftGmmState*, u16*, u16*);
+};
+
+void FieldSystem_InitGetMysteryGiftGmmState(struct GetMysteryGiftGmmState* state, FieldSystem* fieldSys, MessageFormat* msgFormat, MysteryGiftData* mgData);
```

Comment:
if these are static they need the static keyword, if they're extern they need the extern keyword

Reply:
also do not use struct keyword for struct params

Reply:
disagree with the second comment

Reply:
style again, this was majority agreed